osmosis/src/setupStore.js

```diff
@@ -15,6 +15,10 @@ const setupStore = useCustomHook => {
 
   const withStoreContext = WrappedComponent => props => {
     let store = useCustomHook();
+    if (store.Provider != undefined || store.Context != undefined)
+    {
+      throw "cannot have previous 'Provider' or 'Context' defined";
+    }
```

Comment:
Lets use truthy checks here, and also lets check them distinctly and modify the thrown exception so that it is specific to protected property in question. Lets also improve the usability of the error message for our consumers. Something like:
```
if (!!store.Context) throw new Error("'Context' property is protected and cannot exist on a store object");
if (!!store.Provider) throw new Error("'Provider' property is protected and cannot exist on a store object");
```